spec/src/main/asciidoc/appendix.adoc

```diff
@@ -38,6 +38,47 @@ https://github.com/unitsofmeasurement[UoM, JSR 363]
 
 http://metrics20.org/spec/[Metrics 2.0]
 
+
+
+=== Application REST metrics
+
+Metrics gathered from RESTful stats are optional and therefore may not be available in every implementation. These metrics that are created by gathering RESTful stats are to be portable between different implementations and as such are not valid as optional vendor metrics. These metrics belong under the application scope as they are metrics derived from observing the use of RESTful resources in the application.
+
+It is up to the implementation to decide how to enable the REST metrics.
+
+These metrics require that RESTful stats are gathered from all RESTful resources.
+
+
+*(Optional) RESTRequests*
+[cols="1,4"]
+|===
+|Name| rest.request.total
+|DisplayName| Total Requests
+|Type| Counter
+|Unit| None
+|Multi| true
+|Tags| {class=%s1,method=%s2}
+|Description| The number of invocations of this RESTful resource method since the start of the server.
+|Notes|The `%s1` should be substituted with the fully qualified name of the RESTful resource class. 
+  
+The `%s2` should be substituted with the name of the RESTful resource method and appended with its parameter types using an underscore `\_`.  Multiple parameter types are appended one after another (e.g. `<methodName>_<paramType1>_<paramType2>`).
```

Comment:
Maybe we should include some more details how to encode the parameter types?
- should it be a fully qualified class name or the short name? If fully qualified, then dots should be replaced with what? Because if we replace them with underscores and we also have underscores between the different parameter types, it might look a bit confusing (`java_lang_String_java_lang_Integer`). This is not yet an OpenMetrics export so we can include any UTF-8 character.
- should this somehow exclude JAX-RS injected types, like arguments annotated with `@Context`? They are not quite relevant for this (I'm personally not sure about this, perhaps it's not necessary, just an idea)
- how to encode array types, varargs and generic types? (perhaps some of these are not allowed in REST method parameters, I'm not that familiar with it)

Reply:
It would be fully qualified and since it's the tag value we can use java.lang.String and have underscores separate them.

After talking with @andymc12 , the `@Context` annotated params should be included and param rules are:
```
Parameter type formatting rules: +
- The paramter types are fully qualified (e.g. `java.lang.Object`). +
- Array paramter types will be formatted as `paramType[]` (e.g `java.lang.Object[]`). +
- A Vararg parameter will be treated as an array. +
- Generics will be ignored. For example `List<String>` will be formatted as `java.util.List`.
```

Reply:
Cool! Thanks